### DISCUSSION TITLE
keywords seems to have no effect when using tier nova

### DISCUSSION THREAD

**Discussion by ilbrigz** [deepgram/community#262] [General help] [answered]
Does tier=nova support keywords functionality? It seems like the results using `/listen?tier=nova&search=top&keywords=top:10` or `/listen?tier=nova&search=top` makes no difference. 

Only when I remove the tier=nova from both requests does the results starts to make difference and seems to be more accurate.

**Comment by jjmaldonis** [comment]
Hey @ilbrigz thanks for the post and info. We'd like to dig into this in-depth and some additional info from you would be extremely helpful.

1. Do you have a pair of request IDs where the Base model is working but Nova is not? (Base is the model that gets used when tier=nova is not included). You can find the request IDs from the API response or in your Console -> Usage -> Logs.
2. What is the full URL you are using? Is it `https://api.deepgram.com/v1/listen?tier=nova&search=top&keywords=top:10`?
3. Can you please upload your audio file and attach it to your post? GitHub requires you to zip the file before uploading it. If you'd rather email it to me, let me know.
4. Can you please attach the responses of the APIs calls as well, especially if you don't attach the audio file? In particular, we want to look at the confidence values for the word `top` in the response, so this information is critical to see or to reproduce.

Lastly, please note that the `search=top` query parameter will not perform boosting and returns a different API response.

**Reply by ilbrigz** [reply]
HI. I'm actually trying to build an app on which my kid would read a 3-letter word and deepgram would then tell the app if my kid is reading the word right. Using only 3-letter words would make them very common so bots, even humans, would find it difficult sometimes to decide what word is actually being pronounced. That being said, keywords in my opinion would be very important for my app if I am to use deepgram.

To make things simpler I removed the search query from the url. Again, the problem is that keywords are not working when I am using the tier nova. I prefer tier nova because it's much cheaper and I might publish this app online.

The actual spoken words were "fat six pat rid top ten". The audio is not clear becuase I only used the camera microphone so understandably Deepgram would make a few mistakes. Here are the urls, responses, and request ID.  This time I'm trying to tell deepgram that the important word is "ten"

`https://api.deepgram.com/v1/listen?`
fund six pat read top and
#7c7eb3f0-bdea-4a6a-8887-7527da34b437

`https://api.deepgram.com/v1/listen?keywords=ten:10`
find six pat read top ten
#18a416b6-284d-4a17-a6c7-61fcab64d8a0

`https://api.deepgram.com/v1/listen?tier=nova`
find six fat rid tough and
#3183c507-68c9-4bb3-adab-26f1e8b4033a

`https://api.deepgram.com/v1/listen?tier=nova&keywords=ten:10`
find six fat rid tough and
#0e25101e-60ac-44ed-9224-ed6fc46c4c3b

The first 2 request made sense. The last two made no difference.

[audio(11).zip](https://github.com/deepgram/community/files/11969879/audio.11.zip)

**Reply by ilbrigz** [reply, marked as answer]
Actually Sandra_R replied to me in discord. She said that keywords are beta feature and they don't currently work with Nova. So there was nothing wrong above. It's just not a feature yet.

**Comment by cateas** [comment]
@jjmaldonis we have the same issue with the keywords. Is it possible to get an ETA when this issue may get fixed? It would help us a lot, since we have a lot of names in the call recordings which are in most of the cases not transcribed very well.

Thank you!

**Reply by jjmaldonis** [reply]
Hey @cateas I don't have an ETA but I would like to pass more information to our team to speed up the process of getting this fixed. One important distinction is between "in vocab" keywords and "out of vocab" keywords. "In vocab" means the word is in the vocabulary that the model was trained on, and "out of vocab" means the opposite. For ilbrigz's case, the word `top` is in-vocab, but for your case it sounds like the issue is with company names which will be out of vocab. So for your case, it will be very helpful to get more details:
* What are the keywords you are trying to boost?
* How much are they being boosted by?
* What are the confidence scores of the incorrect results in the API response? - i.e. in this part of the response: `response["results"]["channels"][0]["alternatives"][0]["words"][0]["confidence"]`
* If you're able to send me the audio file that would be very helpful as well. This will allow us to rerun your API request, see the issue, and use it as a test case while fixing the issue.

**Reply by cateas** [reply]
@jjmaldonis I will share detailed response tomorrow. Thanks!

**Reply by cateas** [reply]
@jjmaldonis I have shared the files via email.

Looking forward for your feedback. Thanks!
